src/Common/FitOnFhir.Common.Tests/AuthStateServiceTests.cs

```diff
@@ -94,6 +96,15 @@ public void GivenAnonymousLoginEnabledWithValidRequest_WhenCreateAuthStateCalled
             }
         }
 
+        [Fact]
+        public void GivenAnonymousLoginEnabledWithInvalidRequest_WhenCreateAuthStateCalled_ThrowsAuthStateException()
```

Comment:
We are going to need to expand the tests here to explicitly cover all of the cases parameter failure cases. We need to cover all the different permutations of this failure where one or 2 parameters are null, and validate the error message is correct.

Reply:
good old. try/catch.